### PR TITLE
Comments: More comments on functions/globals in standard.h.

### PR DESCRIPTION
**Pull request by jimpo**
I was confused about what "data carrier" meant, so I wanted to comment the `fAcceptDatacarrier` and `nMaxDatacarrierBytes` fields specifically. Then I figured I'd add docs for the rest of the functions.

### CODE REVIEW

File: src/script/standard.h [hunk 1]
```diff
@@ -70,15 +81,49 @@ class CNoDestination {
  */
 typedef boost::variant<CNoDestination, CKeyID, CScriptID> CTxDestination;
 
+/** Get the name of a txnouttype as a C string, or nullptr if unknown. */
 const char* GetTxnOutputType(txnouttype t);
 
+/**
+ * Parse a scriptPubKey and identify script type for standard scripts. Assigns
+ * the result to the typeRet reference parameter and returns true if successful.
```
Comment (TheBlueMatt) [inline src/script/standard.h]: Looks like you lost the stuff noting what vSolutionsRet is when you moved the comment from standard.cpp.

File: src/script/standard.h [hunk 2]
```diff
@@ -27,16 +27,27 @@ class CScriptID : public uint160
...
+ */
+static const unsigned int MAX_OP_RETURN_RELAY = 83;
+
+/**
+ * A data carrying output is an unspendable output containing data. The script
+ * type is designated as TX_NULL_DATA.
+ */
 extern bool fAcceptDatacarrier;
+
+/** Maximum size of TX_NULL_DATA scripts that this node considers standard. */
```
Comment (TheBlueMatt) [inline src/script/standard.h]: (per transaction)

File: src/script/standard.h [hunk 3]
```diff
@@ -70,15 +81,49 @@ class CNoDestination {
...
 const char* GetTxnOutputType(txnouttype t);
 
+/**
+ * Parse a scriptPubKey and identify script type for standard scripts. Assigns
+ * the result to the typeRet reference parameter and returns true if successful.
+ */
 bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::vector<unsigned char> >& vSolutionsRet);
+
+/**
+ * Parse a standard scriptPubKey for the destination address. Assigns result to
```
Comment (TheBlueMatt) [inline src/script/standard.h]: Maybe note that this only works on some standard scriptPubKey types (eg currently does not work on segwit types, nor null_data, for obvious reasons).

**Review by TheBlueMatt — COMMENTED** [review]
Thanks for this!

File: src/script/standard.h (hunk 1 above)
Comment (jnewbery) [inline src/script/standard.h]: (specifically that vSolutionsRet is a vector of the public keys or hashes from the scriptPubKey)

File: src/script/standard.h (hunk 2 above)
Comment (jimpo) [inline src/script/standard.h]: I'm not sure if that's necessary to add here. This is the byte limit for OP_RETURN scripts, then `IsStandardTx` further restricts the number of OP_RETURN outputs per tx to 1. If this comment were to say `per transaction` it sort of implies that the data can be split across multiple OP_RETURN outputs as long as the limit is not exceeded.

File: src/script/standard.h (hunk 3 above)
Comment (jimpo) [inline src/script/standard.h]: Happy to add if you prefer, but that seems like something that the implementation might change. Is it possible that a new `CTxDestination` for witness programs will be added? It's also interesting that this function does not return a `CNoDestination` for `TX_NULL_DATA` scripts.

File: src/script/standard.h (hunk 2 above)
Comment (TheBlueMatt) [inline src/script/standard.h]: Ahh, missed that it was applied per-scriptPubKey and had an independant per-tx limit, yea, fine as-is.

File: src/script/standard.h (hunk 3 above)
Comment (TheBlueMatt) [inline src/script/standard.h]: Sure, but we can also update the comment when the implementation changes to support more :)

**Review by TheBlueMatt — COMMENTED** [review]
ACK 360b464a08ac42a9a249b34c1a780ada2d4f610a

**Comment by laanwj** [comment]
ACK 360b464, thanks for adding documentation

**Pull request merged**